### PR TITLE
Replace "experimental" with "may change"

### PR DESCRIPTION
**Pull request by johanandren**
Refs #22305

### CODE REVIEW

File: akka-docs/rst/common/may-change.rst [hunk 1]
```diff
@@ -0,0 +1,49 @@
+
+.. _may-change:
+
+###########################
+Modules marked "May Change"
+###########################
+
+To be able to introduce new modules and APIs without freezing them the moment they
+are released because of our :ref:`BinCompatRules <binary compatibility guarantees>` we have introduced
```
Comment (ktoso) [inline akka-docs/rst/common/may-change.rst]: no short wording in normal text, "Binary Compatibility Rules" instead

Comment (johanandren) [inline akka-docs/rst/common/may-change.rst]: um, I thought "binary compatibility guarantees" is the text in the link, did I get the link syntax wrong?

Comment (johanandren) [inline akka-docs/rst/common/may-change.rst]: Yep I did.

Comment (ktoso) [inline akka-docs/rst/common/may-change.rst]: Hm I think <> is the target? Double check though...

Comment (johanandren) [inline akka-docs/rst/common/may-change.rst]: (I hate you, re-structured-text!)

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -300,11 +300,12 @@ You can read up on `remaining` and friends in [TestKit.scala](https://github.com
 
 For external contributions of entire features, the normal way is to establish it
 as a stand-alone feature first, to show that there is a need for the feature. The
-next step would be to add it to Akka as an "experimental feature" (in the
-akka-contrib subproject), then when the feature is hardened, well documented and
+next step would be to add it to Akka as an "may change"-feature (in the
+akka-contrib subproject) and marking it's public api with the `MayChange` annotation,
```
Comment (johanandren) [inline CONTRIBUTING.md]: Ups, `ApiMayChange` is the annotation

File: akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst [hunk 3]
```diff
@@ -512,3 +512,16 @@ the Agents, as they rarely are really enough and do not fit the Akka spirit of t
 We also anticipate to replace the uses of Agents by the upcoming Akka Typed, so in preparation thereof the Agents have been deprecated in 2.5.
 
 If you use Agents and would like to take over the maintanance thereof, please contact the team on gitter or github.
+
+Akka Typed
+==========
+
+With the new term :ref:`may change <may-change>` we will no longer have a different artifact for modules that are not
+stable, and ``akka-typed-experimental`` has therefore been renamed to ``akka-typed``. Note that it is still promoted to
+a stable module.
```
Comment (patriknw) [inline akka-docs/rst/project/migration-guide-2.4.x-2.5.x.rst]: still **not** promoted

**Review by patriknw — APPROVED** [review]
LGTM after one important change

**Comment by drewhk** [comment]
What about artifact names? They will look like `akka-thing-may-change.jar`?

**Comment by johanandren** [comment]
Nope, no they will no longer contain the marker (that is what I talked about during the morning meeting), just in their description and docs.

**Comment by drewhk** [comment]
Ok, I see

File: akka-docs/rst/common/binary-compatibility-rules.rst [hunk 4]
```diff
@@ -80,25 +80,11 @@ thus such change is considered safe.
...
-An experimental module doesn't have to obey the rule of staying binary
-compatible between micro releases. Breaking API changes may be introduced
-in minor releases without notice as we refine and simplify based on your
-feedback. An experimental module may be dropped in minor releases without 
-prior deprecation.
-
-Best effort migration guides may be provided, but this is decided on a case-by-case basis for **experimental** modules.
+The meaning of "may change"
+===========================
+**may change** is used in module descriptions and docs in order to signify that the API that they contain
```
Comment (ktoso) [inline akka-docs/rst/common/binary-compatibility-rules.rst]: Uppercase M

**Pull request merged**